addon/components/flash-message.js

```diff
@@ -58,14 +58,16 @@ export default Component.extend({
       set(this, 'active', true);
     });
     set(this, 'pendingSet', pendingSet);
-    this.element.addEventListener('mouseenter', this._mouseEnter);
-    this.element.addEventListener('mouseleave', this._mouseLeave);
+    this.set('_mouseEnterHandler', this._mouseEnter.bind(this));
+    this.set('_mouseLeaveHandler', this._mouseLeave.bind(this));
```

Comment:
Should we approach with 1 format here instead of having a `this.set` and `set(this, ...)` mixed in code? In fact it would be nicer to use `Ember.setProperties(this, {})`. Thoughts? @Turbo87 @sbatson5

Reply:
since `_mouseEnterHandler` is never observed in the template you can also just use `this._mouseEnter = this._mouseEnter.bind(this);` to simplify it a little bit

Reply:
I'd lean toward `Ember.set` since that's the convention used elsewhere.

Reply:
@sbatson5 I'm confused: you 👍ed @Turbo87's suggestion and proposed another alternative. Which one would you like to be implemented?

Reply:
@dcyriller Sorry for the confusion.  I was 👍ing the suggestion that `this.foo = x` is possible rather than `this.set('foo', x)`.  However, given that that this repo is still using `Ember.set('foo', x)` everywhere, I would lean toward that approach.